src/neo/Ledger/Blockchain.cs

```diff
@@ -431,14 +431,19 @@ private void OnParallelVerified(ParallelVerified parallelVerified)
             RelayResultReason reason = parallelVerified.VerifyResult;
             if (reason == RelayResultReason.Succeed)
             {
-                if (View.ContainsTransaction(parallelVerified.Transaction.Hash))
-                    reason = RelayResultReason.AlreadyExists;
-                else if (!MemPool.CanTransactionFitInPool(parallelVerified.Transaction))
-                    reason = RelayResultReason.OutOfMemory;
-                else if (!MemPool.TryAdd(parallelVerified.Transaction.Hash, parallelVerified.Transaction))
-                    reason = RelayResultReason.OutOfMemory;
-                else if (parallelVerified.ShouldRelay)
-                    system.LocalNode.Tell(new LocalNode.RelayDirectly { Inventory = parallelVerified.Transaction });
+                reason = parallelVerified.Transaction.VerifyForEachBlock(currentSnapshot, MemPool.SendersFeeMonitor.GetSenderFee(parallelVerified.Transaction.Sender));
```

Comment:
```cs
            if (hashes.Length != Witnesses.Length) return RelayResultReason.Invalid;
            for (int i = 0; i < hashes.Length; i++)
            {
                if (Witnesses[i].VerificationScript.Length > 0) continue;
                if (snapshot.Contracts.TryGet(hashes[i]) is null) return RelayResultReason.Invalid;
            }
```

Maybe some of these could be avoided, at least the first line.
But it is a just a minor thing.

I am not sure if the snapshot could contain a previous contract that is now deleted.

Reply:
If we remove the first line, we can have an exception